src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,150 @@
+
+
+    /**
+     * This method returns a user {@link UserBook}, which match with username received as parameter
+     *
+     * @param username user's username
+     *
+     * @return {@link Optional} book's user
+     */
+    @GetMapping("/user/{userUsername}")
+    public Optional<UserBook> findByUsername(@PathVariable String username) {
+        return userRepository.findByUsername(username);
+    }
+
+    /**
+     * This method returns a user {@link UserBook}, which match with name received as parameter
+     *
+     * @param name user's name
+     *
+     * @return {@link Optional} book's user
+     */
+    @GetMapping("/user/{name}")
+    public Optional<UserBook> findByName(@PathVariable String name) {
+        return userRepository.findByName(name);
+    }
+
+    /**
+     * This method creates a user object
+     *
+     * @param user: user's parameters
+     *
+     * @return {@link Book} user created
+     */
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public UserBook create(@RequestBody UserBook user) {
+        return userRepository.save(user);
+    }
+
+    /**
+     * This method deleted a user, which match with id received by parameter
+     *
+     * @param id: User's identify number
+     */
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
+        userRepository.findById(id)
+                .orElseThrow(() -> new UserNotFoundException("Not found user id"));
+        userRepository.deleteById(id);
+    }
+
+    /**
+     * This method deleted a book from user's list books, which match with id received by parameter
+     *
+     * @param idUser: User's identify number
+     * @param idBook: Book's identify number
+     */
+    @DeleteMapping("/{idUser}/{idBook}")
+    public void deleteBookListUser(@PathVariable String idUser, @PathVariable String idBook) {
+
+        UserBook user = userRepository.findById(Long.parseLong(idUser))
+                .orElseThrow(() -> new UserNotFoundException("Not found user id"));
+        Book book = bookRepository.findById(Long.parseLong(idBook))
+                .orElseThrow(() -> new UserNotFoundException("Not found book id"));
+        List<Book> books = user.deleteBookList(book);
+        user.setBooks(books);
+        userRepository.save(user);
+    }
+
+    /**
+     * This method add a book from user's list books, which match with id received by parameter
+     *
+     * @param idUser: User's identify number
+     * @param idBook: Book's identify number
+     */
+    @PutMapping("/{idUser}/{idBook}")
+    public void addBookListUser(@PathVariable String idUser, @PathVariable String idBook) {
+
+        UserBook user = userRepository.findById(Long.parseLong(idUser))
+                .orElseThrow(() -> new UserNotFoundException("Not found user id"));
+        Book book = bookRepository.findById(Long.parseLong(idBook))
+                .orElseThrow(() -> new UserNotFoundException("Not found book id"));
```

Comment:
You are calling the same exception for Users and Books, I know both are NotFoundExceptions but you should throw a BookNotFoundException for Books errors.